src/content-scanner/ContentScanner.ts

```diff
@@ -86,52 +86,38 @@ export class ContentScanner {
             return this.pendingScans.get(mxc);
         }
 
-        // eslint-disable-next-line no-async-promise-executor
-        const promise = new Promise<boolean>(async (resolve, reject) => {
-            let response: Response;
-
-            if (file) {
-                if (!this.hasKey) {
-                    try {
-                        const k = await fetch(this.scannerUrl + "/_matrix/media_proxy/unstable/public_key")
-                            .then(r => r.json());
-                        this.mcsKey.set_recipient_key(k["public_key"]);
-                        this.hasKey = true;
-                    } catch (err) {
-                        reject(err);
-                        return;
-                    }
-                }
-
-                try {
-                    response = await fetch(this.scannerUrl + "/_matrix/media_proxy/unstable/scan_encrypted", {
-                        method: "POST",
-                        body: JSON.stringify({
-                            encrypted_body: this.mcsKey.encrypt(JSON.stringify({ file })),
-                        }),
-                        headers: {
-                            "Content-Type": "application/json",
-                        },
-                    });
-                } catch (err) {
-                    reject(err);
-                    return;
-                }
-            } else {
-                const url = this.scannerUrl + `/_matrix/media_proxy/unstable/scan/${mxc.substring('mxc://'.length)}`;
-                try {
-                    response = await fetch(url);
-                } catch (err) {
-                    reject(err);
-                    return;
-                }
+        const prom = this.doScan(mxc, file);
+        this.pendingScans.set(mxc, prom);
```

Comment:
"pendingScans" is a bit inaccurate : it also contains finished scans. (I just realized that finished scans are not removed, so this doesn't do what I thought!)

Maybe "startedScans" or "alreadyScanned" or "cachedScans"?

Reply:
Renamed to `cachedScans`